### ISSUE TITLE
Feature request: sh access

### ISSUE THREAD

**Issue by sarink** [jesseduffield/lazydocker#35]
Next to the tabs `Logs | Stats | Container Config | Top` I was thinking it would be pretty neat to have one called `shell`, when you click it it could execute `docker-compose <container> exec sh` and that box can become a repl?

**Comment by mudasar187** [comment]
> Next to the tabs `Logs | Stats | Container Config | Top` I was thinking it would be pretty neat to have one called `shell`, when you click it it could execute `docker-compose <container> exec sh` and that box can become a repl?
> 
> BTW, thank you for this project 🎊 I've been using it for a day and it's amazing!

That would be amazing to have! :D

**Comment by jesseduffield** [comment]
currently if you hit 'c' it brings up custom commands, of which one is the `sh` command, however that switches you into a fullscreen bash prompt. I really would love to have stuff like bash run in the main panel on the right, but it's actually quite a tricky problem to solve, given that you basically need to get a pseudo terminal and then interpret all of the ansi escape codes in order to render the ascii characters in the panel. But it is probably top of my long-term priorities after supporting podman

**Comment by markphip** [comment]
> currently if you hit 'c' it brings up custom commands, of which one is the `sh` command, however that switches you into a fullscreen bash prompt. I really would love to have stuff like bash run in the main panel on the right, but it's actually quite a tricky problem to solve, given that you basically need to get a pseudo terminal and then interpret all of the ansi escape codes in order to render the ascii characters in the panel. But it is probably top of my long-term priorities after supporting podman

FWIW, this works pretty well.  Might just be nice if there were a direct shortcut for this specific option (to get a shell)? This is not bad at all though as is.

**Comment by jesseduffield** [comment]
@markphip I've made an issue for this https://github.com/jesseduffield/lazydocker/issues/101. I'm aware that it's not strictly necessary for this thing to be a custom command, though I think it's a good basis for showing users how they can configure this stuff themselves, by setting some exemplar defaults

**Comment by markphip** [comment]
Makes sense. I was mainly commenting on the shell behavior though. I find what you are currently doing works pretty well in practice.

**Comment by jesseduffield** [comment]
good to hear :)

**Comment by gggeek** [comment]
Btw: I tried getting shell access into the lazydocker container itself, and it failed with error message:

OCI runtime exec failed: exec failed: container_linux.go:346: starting container process caused "exec: \"/bin/sh\": stat /bin/sh: no such file or directory"

It seems to me an useful feature to be able to drop to a shell into the lazydocker container, to be able to snoop around, learn and troubleshoot. Shall I open a separate bug/feature request ?

**Comment by jesseduffield** [comment]
@gggeek sounds like `/bin/sh` isn't actually a program defined in your container. Not sure how best to get around that

**Comment by etiedt** [comment]
I would also love to see this feature
